components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -340,16 +339,19 @@ handler.on(PanelActions.signMessageHardware.getType(), async (store, messageData
   await navigateToConnectHardwareWallet(store)
   const info = hardwareAccount.hardware
   const signed = await signMessageWithHardwareKeyring(apiProxy, info.vendor, info.path, messageData.message)
-  if (!signed.success &&
-      (signed.code === TrezorErrorsCodes.CommandInProgress ||
-       signed.code === LedgerErrorsCodes.TransportLocked)) {
-      // do nothing as the operation is already in progress
+  if (!signed.success) {
+    if (signed.code && (signed.code !== LedgerStatusCodes.CONDITIONS_OF_USE_NOT_SATISFIED ||
+                        signed.code === TrezorErrorsCodes.CommandInProgress)) {
+      const deviceError = dialogErrorFromLedgerErrorCode(signed.code)
```

Comment:
As for `signed.code !== LedgerStatusCodes.CONDITIONS_OF_USE_NOT_SATISFIED`, you can instead do `deviceError !== 'transactionRejected'`.

Regarding `signed.code === TrezorErrorsCodes.CommandInProgress`, is it intended that this be passed to `dialogErrorFromLedgerErrorCode`? I was thinking it would be better to separate vendor-specific calls, even though it results in some code duplication.

Reply:
done